serverless-functions/src/functions/common/twilio-wrappers/sync.private.js

```diff
@@ -4,6 +4,123 @@ const retryHandler = require(Runtime.getFunctions()[
   "common/twilio-wrappers/retry-handler"
 ].path).retryHandler;
 
+/**
+ * @param {object} parameters the parameters for the function
+ * @param {number} parameters.attempts the number of retry attempts performed
+ * @param {object} parameters.context the context from calling lambda function
+ * @param {string} parameters.mapSid the SID of the Sync Map
+ * @param {string} parameters.key the key of the Sync Map item
+ * @returns {object} A new Sync Map Item
+ * @description the following method is used to remove a Sync Map Item
+ */
+ exports.deleteMapItem = async function (parameters) {
+  const { attempts, context, mapSid, key } = parameters;
+
+  if (!isNumber(attempts))
+    throw "Invalid parameters object passed. Parameters must contain the number of attempts";
+  if (!isObject(context))
+    throw "Invalid parameters object passed. Parameters must contain context object";
+  if (!!mapSid && !isString(mapSid))
+    throw "Invalid parameters object passed. Parameters must contain context object";
+  if (!!key && !isString(key))
+    throw "Invalid parameters object passed. Parameters must contain uniqueName string value";
+
+  try {
+    const client = context.getTwilioClient();
+
+    const mapItem = await client.sync
+      .services(context.TWILIO_FLEX_SYNC_SID)
+      .syncMaps(mapSid)
+      .syncMapItems(key).remove();
+
+    return { success: true, status: 200, mapItem: mapItem };
```

Comment:
I don't think a map item is returned by the API so you might not need to return anything other than success/status.